engine-nextgen/src/main/kotlin/graphql/nadel/NextgenEngine.kt

```diff
@@ -273,6 +293,24 @@ class NextgenEngine @JvmOverloads constructor(
         )
     }
 
+    private fun usesDynamicServiceResolution(
+        topLevelField: ExecutableNormalizedField
+    ): Boolean =
+        topLevelField.getFieldDefinitions(overallSchema)
+            .asSequence()
+            .filter {
+                it.getDirective(NadelDirectives.DYNAMIC_SERVICE_DIRECTIVE_DEFINITION.name) != null
+            }
+            .onEach {
+                Assert.assertTrue(GraphQLTypeUtil.unwrapNonNull(it.type) is GraphQLInterfaceType) {
+                    String.format(
+                        "field annotated with %s directive is expected to be of GraphQLInterfaceType",
+                        NadelDirectives.DYNAMIC_SERVICE_DIRECTIVE_DEFINITION.name
+                    )
+                }
+            }
+            .firstOrNull() != null
```

Comment:
just FYI, don't have to change: you can use `.any()` to check if a sequence is empty or not

Reply:
nice ;-)

Reply:
interestingly, if I use `any` the code block in `onEach` doesn't execute and errors are not thrown when expected, strange.

Reply:
Weird, maybe you need a `filterNotNull().any()`?